### PR TITLE
Fix Sampler.ShouldSample parameter descriptions.

### LINKED ISSUE #883: Sampler.ShouldSample has wrong parameter descriptions.
Late follow-up from #634.

In https://github.com/open-telemetry/opentelemetry-specification/blob/14b5b6a/specification/trace/sdk.md#shouldsample, the spec has some confusing or even wrong wording when it describes the parameters, namely:

* > `SpanContext` of a parent Span.:
  + > Typically extracted from the wire.

    This is not true, since samplers are called for all child spans, not only for (local) root spans. A parent is often a locally created Span.
  + > Can be null.

    Normally we use an empty/invalid SpanContext everywhere instead of null. This was actually changed in Java in https://github.com/open-telemetry/opentelemetry-java/issues/1487

* > TraceId of the Span to be created. It can be different from the TraceId in the SpanContext. Typically in situations when the Span to be created starts a new Trace.

  I don't think this can be different. If there is any valid parent trace ID, then it must match the newly created Span. If the Span where to start a new Trace, the parent context would be empty (or null with the current wording).

Marking this as P3 since I think the text in question is not really normative. But it's still wrong, so I think it would be desirable to fix this before GA.

CC @alolita

### PR DESCRIPTION
**Pull request by Oberon00**
Fixes #883.

Only non-normative changes, so no CHANGELOG entry.

### CODE REVIEW

File: specification/trace/sdk.md [hunk 1]
```diff
@@ -60,14 +60,12 @@ Returns the sampling Decision for a `Span` to be created.
 
 **Required arguments:**
 
-* `SpanContext` of a parent `Span`. Typically extracted from the wire. Can be
-  `null`.
-* `TraceId` of the `Span` to be created. It can be different from the `TraceId`
-  in the `SpanContext`. Typically in situations when the `Span` to be created
-  starts a new Trace.
+* Parent `SpanContext`. May be invalid to indicate a root span.
```
Comment (carlosalberto) [inline specification/trace/sdk.md:63]: Either `null` or `invalid`?

File: specification/trace/sdk.md [hunk 2]
```diff
@@ -60,14 +60,12 @@ Returns the sampling Decision for a `Span` to be created.
...
+* `TraceId` of the `Span` to be created.
+  If the parent `SpanContext` contains a valid `TraceId`, they always match.
```
Comment (carlosalberto) [inline specification/trace/sdk.md:65]: they MUST always match?

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
Overall fine. Left a pair of small, non-blocking comments.

File: specification/trace/sdk.md (hunk 1 above)
Comment (Oberon00) [reply specification/trace/sdk.md:63]: I don't think we allow null anywhere else, so I intentionally removed that.

File: specification/trace/sdk.md (hunk 2 above)
Comment (Oberon00) [reply specification/trace/sdk.md:65]:
```suggestion
  If the parent `SpanContext` contains a valid `TraceId`, they MUST always match.
```

**Review by arminru — APPROVED** [review]

**Comment by SergeyKanzhelev** [comment]
can you please rebase so this can be merged?

image

**Comment by arminru** [comment]
@SergeyKanzhelev I thought this setting was changed in the repo following https://github.com/open-telemetry/opentelemetry-specification/issues/512#issuecomment-616650793? If there are no conflicts, merging should work fine even if another unrelated PR was merged after the last merge/rebase.

**Comment by SergeyKanzhelev** [comment]
> @SergeyKanzhelev I thought this setting was changed in the repo following [#512 (comment)](https://github.com/open-telemetry/opentelemetry-specification/issues/512#issuecomment-616650793)? If there are no conflicts, merging should work fine even if another unrelated PR was merged after the last merge/rebase.

This is why I posted the screenshot. It didn't even gave me a button "Update branch". Perhaps a GitHub glitch. Merging now as it is widely approved and uncontroversial.

**Pull request merged**